## 1.	Print the numbers from 0 to 5
using for, using while

In [1]:
# using for
for i in range(0,6):
    print(i)


0
1
2
3
4
5


In [2]:
# using while
i =0
while i<6:
    print(i)
    i+=1


0
1
2
3
4
5


In [3]:
# Try the same in list comprehension (not advisable to use list comprehension just to print numbers)
# (Just for practice purpose)

[ i for i in range(0,6)]

[0, 1, 2, 3, 4, 5]

## 2.	Print all the even numbers between 22 and 75

In [4]:
## for loop
for i in range(22,76):
    if i%2==0:
        print(i)

22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74


In [5]:
## while
i =22
while (i>=22) & (i<76):
    if i%2==0:
        print(i)
    i +=1

22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74


In [6]:
## Using list comprehension 
[i for i in range(22,76) if i%2 ==0]

[22,
 24,
 26,
 28,
 30,
 32,
 34,
 36,
 38,
 40,
 42,
 44,
 46,
 48,
 50,
 52,
 54,
 56,
 58,
 60,
 62,
 64,
 66,
 68,
 70,
 72,
 74]

## 3.	Print all the letters in a string, tuple, list and dictionary

In [7]:
c = "Python"
for i in c:
    print(i)

P
y
t
h
o
n


In [8]:
d = ("great","learning","greatlearning")
for i in d:
    print(i)

great
learning
greatlearning


In [9]:
e = ["great","learning","greatlearning"]
for i in e:
    print(i)

great
learning
greatlearning


In [10]:
f =  {"great":26,"learning":23,"greatlearning":28}
## only keys
keys =f.keys()
    #print(v)

print(keys)    
## only values
value = f.values()
print(value)    
    
## both values and keys
items =f.items()
print(items)

dict_keys(['great', 'learning', 'greatlearning'])
dict_values([26, 23, 28])
dict_items([('great', 26), ('learning', 23), ('greatlearning', 28)])


In [11]:
for i,v in f.items():
    print(i,v)
    

great 26
learning 23
greatlearning 28


## 4.	Print the total sum and cumulative sum of all the numbers in the list o = [6, 5, 3, 8, 4, 2, 5, 4, 11]

In [12]:
g = [6, 5, 3, 8, 4, 2, 5, 4, 11]

## sum
sum1 =0
for i in g:
    sum1 = sum1+i
print(sum1)
    

## cumsum 
print('cummulative sum ')
sum1 =0
for i in g:
    sum1 = sum1+i
    print(sum1)

48
cummulative sum 
6
11
14
22
26
28
33
37
48


In [13]:
## practice the same with advance functions 

# sum--reduce
from functools import reduce
print(reduce(lambda a,b : a+b,g))
# Cumsum--accumulate
from itertools import accumulate
a =list(accumulate(g,lambda a,b : a+b))
a
#itertools.accumulate

48


[6, 11, 14, 22, 26, 28, 33, 37, 48]

## 5.	Combine the given lists a & b, get the count of elements. Eg. ‘hello’ Occurs 3 times, ‘barbie’ occurs 2 times etc.,
a=['hello','world','barbie','hello']
b=['hello','barbie','hai']


In [14]:
a=['hello','world','barbie','hello'] 
b=['hello','barbie','hai']

c = a+b
d =set(c)
for i in d:
    print(i,c.count(i))
#print(cc)

world 1
hai 1
hello 3
barbie 2


## 6.	Write a program to remove all the even numbers from the given set, print the updated set and the removed numbers in a new set.
S1 = {7,14,21,28,35,42,49,56,63,70,77,84}


In [15]:
S1 = {7,14,21,28,35,42,49,56,63,70,77,84}
S3 = S1.copy()
S2 =set()
for i in S3:
    if i%2==0:
        
        S2.add(i)
        S1.remove(i)
print(S1)
print(S2)


{35, 7, 77, 49, 21, 63}
{70, 42, 14, 84, 56, 28}


## 7.	Write a function to check if the given number is prime or not

In [16]:
def primenum(a):
    lst =[]
    for i in range(2,a):
        
        if a%i==0:
            lst.append(i)
        else:
            pass
    return len(lst)
    #if len(lst)>0:
        
        #print('not prime',len(lst))
        
for i in range(2,100):
    length =primenum(i)
    if length>0:
        print(i,'is notprime')
    else:
        print(i,' is prime')
    #if a%1==0 and a%a==0

2  is prime
3  is prime
4 is notprime
5  is prime
6 is notprime
7  is prime
8 is notprime
9 is notprime
10 is notprime
11  is prime
12 is notprime
13  is prime
14 is notprime
15 is notprime
16 is notprime
17  is prime
18 is notprime
19  is prime
20 is notprime
21 is notprime
22 is notprime
23  is prime
24 is notprime
25 is notprime
26 is notprime
27 is notprime
28 is notprime
29  is prime
30 is notprime
31  is prime
32 is notprime
33 is notprime
34 is notprime
35 is notprime
36 is notprime
37  is prime
38 is notprime
39 is notprime
40 is notprime
41  is prime
42 is notprime
43  is prime
44 is notprime
45 is notprime
46 is notprime
47  is prime
48 is notprime
49 is notprime
50 is notprime
51 is notprime
52 is notprime
53  is prime
54 is notprime
55 is notprime
56 is notprime
57 is notprime
58 is notprime
59  is prime
60 is notprime
61  is prime
62 is notprime
63 is notprime
64 is notprime
65 is notprime
66 is notprime
67  is prime
68 is notprime
69 is notprime
70 is notprime
71  is prim

## 8.	Write a function to print the cube of 78

In [17]:
def cube(a):
    ac = a**3
    return ac
cube(4)


64

In [18]:
## same using lambda

s =lambda x:x**3
s(3)


27

## 9.	Function to check if the given number is a power of 3


In [56]:
def pow_3(a):
    #b =a**(1/3)
    if (round(a**(1/3))%1)==0:
        print('number is perfect cube')
    
pow_3(1000)

number is perfect cube


In [55]:
round(1000**(1/3))

10

## 10.	Function to find factorial of a number (say 6)

In [20]:
def fact(num):
    c =1
    for i in range(1,num+1):
        c =c*i
    return c
fact(4)

24

## 11.	Function to print the biggest of 3 number

In [21]:
def big(a,b,c):
    if a>b &b>c:
        print('a is biggest')
    elif b>a & b>c:
         print('b is biggest')  
    else:
        print('c is biggest')
big(5,4,8)

c is biggest


## 12.	Function to count the occurrences of each word in a given sentence

In [22]:
a = "Nory was a Catholic because her mother was a Catholic"
b =a.split()
c = set(b)
print(c)
#b.count()
for i in c:
    coun =b.count(i)
    print(i,coun)

{'was', 'mother', 'her', 'Nory', 'a', 'Catholic', 'because'}
was 2
mother 1
her 1
Nory 1
a 2
Catholic 2
because 1


## 13.	Customer segmentation, if cust_credit is b/w 5k – 15k mark as silver customer, if it is greater than or equal to 25k, mark as diamond customer, btw 15k and 25k, gold else bronze

In [23]:
def cust(a):
    if (a>5000) & (a<15000):
        print('silver cistomer')
    elif a>=25000:
        print('Diamond cust')
    elif (a>=15000) &(a<25000):
        print('gold cust')
    else:
        print('bronze cust')
cust(25000)    

Diamond cust


## 14.	Count the number of numbers between 555 and 1111 which are divisible by 7

In [24]:
count =0
for i in range(555,1112):
    
    if i%7==0:
        count = count+1
print(count)

79


In [25]:
## using list comprehension
#lst =[]
b =[i for i in range(555,1112) if i%7==0]
len(b)

79

## 15.	Write a program to print list of numbers which are squares between 1 to 30

In [26]:
lst =[]
for i in range(1,31):
    if (i**(1/2)%1)==0:
        lst.append(i)
        #print('number is perfect Square')
print(lst)

[1, 4, 9, 16, 25]


In [27]:
## using list comprehension
[i for i in range(1,31) if (i**(1/2)%1)==0]

[1, 4, 9, 16, 25]

## 16.	for a given range of values 110,130. Print the values and print a message “id 124 is found” in place of 124.

In [28]:
for i in range(110,131):
    if i==124:
        print('id 124 is found')
    else:
        print(i)

110
111
112
113
114
115
116
117
118
119
120
121
122
123
id 124 is found
125
126
127
128
129
130


## 17.	For the given range of value 1 to 10, print all the values till 5 and exit the loop if value 6 is found

In [29]:
for i in range(1,11):
    if i==6:
        break
    else:
        print(i)

1
2
3
4
5


## 18.	Print numbers from 1 to 20. If a multiple of 5 occurs, skip the number. (Output: 1,2,3,4,6,7,8,9,11…..)

In [30]:
## Using For
for i in range(1,21):
    if i%5==0:
        continue
    else:
        print(i)

1
2
3
4
6
7
8
9
11
12
13
14
16
17
18
19


In [31]:
## Using While
i=0
while i<21:
    i+=1
    if i%5==0:
        continue
    else:
        print(i)
    
    
    

1
2
3
4
6
7
8
9
11
12
13
14
16
17
18
19
21


## 19.	Print the list of numbers whose length is greater than or equal to 3
lis = [234, 45, 4, 44, 454, 32445]


In [32]:
## using loops
lis = [234, 45, 4, 44, 454, 32445]
for i in lis:
    if len(str(i)) >=3:
        
        print(i)
    

234
454
32445


In [33]:
## using advance functions (filter, map, lambda)
a =list(filter(lambda i:len(str(i))>=3 ,lis))
a

[234, 454, 32445]

In [34]:
## using advance functions (filter, map, lambda)
list(map(lambda i:len(str(i))>=3 ,lis))
print(a)


[234, 454, 32445]


In [35]:
[lambda i:len(str(i))>=3 for i in lis]

[<function __main__.<listcomp>.<lambda>(i)>,
 <function __main__.<listcomp>.<lambda>(i)>,
 <function __main__.<listcomp>.<lambda>(i)>,
 <function __main__.<listcomp>.<lambda>(i)>,
 <function __main__.<listcomp>.<lambda>(i)>,
 <function __main__.<listcomp>.<lambda>(i)>]

## 20.	Program to print 5 tables using list comprehension (5 x 1 = 5, 5 x 2 = 10, …..)

In [36]:
[5*i for i in range(1,20)]

[5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95]

## 21.	Write a program to filter the odd and even numbers and display them in separately
l = [7,3,8,2,5,9,12,78]


In [1]:
l = [7,3,8,2,5,9,12,78]

print(list(filter(lambda i:i%2==0,l)))
print(list(filter(lambda i:i%2!=0,l)))

[8, 2, 12, 78]
[7, 3, 5, 9]


## 22. Using list comprehension generate the squares of even numbers and the cubes of odd numbers between 1 to 10

In [38]:
b =[i**3 if i%2!=0 else i**2 for i in range(1,11) ]
b 

[1, 4, 27, 16, 125, 36, 343, 64, 729, 100]

## 23.	Write a function to increment all the elements in the list by 2
l = [7,3,8,2,5,9,12,78]

#### 1. Increment 2 without loops
#### 1.Find the sum of all numbers in the list l using reduce function. 
#### 2.Print the cumulative sum of elements in list l using accumulate function.


In [40]:
## using advance functions
#l =[7,3,8,2,5,9,12,78]


In [41]:
l =[7,3,8,2,5,9,12,78]

from functools import reduce
reduce(lambda x,y:x+y ,l)

124

In [42]:
from itertools import accumulate
list(accumulate(l,lambda x,y:x+y))

[7, 10, 18, 20, 25, 34, 46, 124]

## 24.	Create a list d = a + b – c (using map/filter function)
a = [1, 2, -3],
b = [17, -12, 18],
c = [-1, -4, 5],

output: [19, -6, 10]


In [43]:
a = [1, 2, -3]
b = [17, -12, 18]
c = [-1, -4, 5]

list(map(lambda x,y,z:x+y-z,a,b,c))

[19, -6, 10]

In [44]:
#list(filter(lambda x,y,z:x+y-z,(a,b,c)))

## 25.	Create dictionary where the keys are numbers between 1 and 15 (both included) and the values are square of keys

In [45]:
## using dictionary comprehension
a =[(i,i**2) for i in range(1,16)]
dict(a)

{1: 1,
 2: 4,
 3: 9,
 4: 16,
 5: 25,
 6: 36,
 7: 49,
 8: 64,
 9: 81,
 10: 100,
 11: 121,
 12: 144,
 13: 169,
 14: 196,
 15: 225}

In [46]:
## modify the above code, keys should start from 200
a =[(i+199,i**2) for i in range(1,16)]
dict(a)

{200: 1,
 201: 4,
 202: 9,
 203: 16,
 204: 25,
 205: 36,
 206: 49,
 207: 64,
 208: 81,
 209: 100,
 210: 121,
 211: 144,
 212: 169,
 213: 196,
 214: 225}

## 26 . Write a program to filter the numbers from the following string.

In [47]:
a = "gr3e7l34rn1ng"
lst =[]
lst1 =[]
for i in a:
    if i.isnumeric():
        lst.append(i)
    else:
        lst1.append(i)
s =''        
for i in lst:
    s =s+i
print(s)
l =''        
for i in lst1:
    l =l+i
print(l)



37341
grelrnng


## 27. ## Write a program that accepts a sentence and calculate the number of letters and digits.
Suppose the following input is supplied to the program:
hello world 123
Then, the output should be:
LETTERS 10
DIGITS 3

In [48]:
def alpha(c1):
    count =0
    c=0
    print(c1)
    for i in c1:
        if i.isnumeric()==True:
            count =count+1
        elif i.isalpha()==True:
            c =c+1
    return c,count

c1 =input()            
c,count = alpha(c1)
print('letters',c)
print('Digits',count)

hello world 123
hello world 123
letters 10
Digits 3


## 28. Write a program to find out the largest word in the sentence input from user. 
Eg - "This is an umbrella". should give output as umbrella

In [49]:

def big(c):
    value =0
    big =0
    temp =0
    for i in c :
        
        #print(i)
        b = len(i)
        #print(b)
        if b>temp:
            big =b
            #print(big)
            temp =b
            #print(temp)
            value =i
            #print(value)
        else:
            pass
    print(big)
    print(value)
a =input('enter the text')

c =a.split()
print(c)
big(c)

enter the textThis is an umbrella
['This', 'is', 'an', 'umbrella']
8
umbrella


## 29. Find the first 7-digit number that’s divisible by 137? 

In [50]:
lst =[]
for i in range(1,10**8):
    if i%137==0:
        
        if len(str(i))==7:
            
            print(i)
            break

    

1000100


In [62]:
for i in range(1,8):
    if i<=4:
        print(i*'*')
    elif i>4 &i<7:
        print((8-i)*'*')

*
**
***
****
***
**
*


In [63]:
for i in range(1,8):
    if i<=4:
        print(i*'*')

*
**
***
****


In [85]:
for i in range(1,6):
    for j in range(1,i):
        #print(i,end=' ')
        #print(j)
        print(j,end='')
        #print('\n')
    print('\n')



1

12

123

1234



In [102]:
for i in range(1,5):
    for j in range(1,i+1):
        
        print(i,end ='')
    print('\n')   

1

22

333

4444



In [90]:
for i in range(1,6):
    for j in range(i,1,-1):
        #print(i,end=' ')
        #print(j)
        print(j-1,end='')
        #print('\n')
    print('\n')



1

21

321

4321

